include/openPMD/backend/BaseRecordComponent.hpp

```diff
@@ -79,10 +119,34 @@ class BaseRecordComponent : public LegacyAttributable
     ChunkTable
     availableChunks();
 
-    OPENPMD_protected : BaseRecordComponent();
+protected:
+
+    std::shared_ptr< internal::BaseRecordComponentData >
+        m_baseRecordComponentData{
+            new internal::BaseRecordComponentData() };
+
+    inline internal::BaseRecordComponentData const & get() const
+    {
+        return *m_baseRecordComponentData;
+    }
+
+    inline internal::BaseRecordComponentData & get()
+    {
+        return const_cast< internal::BaseRecordComponentData & >(
+            static_cast< BaseRecordComponent const * >( this )->get() );
```

Comment:
For these getters, wouldn't `return *m_baseRecordComponentData;` work here as well and be more legible? :)

I think duplicating the single line does no harmas they are close by and is more readable then the current construct.

Reply:
Yeah .. maybe you're right about this :D 
bit of tunnel vision on my side there

Reply:
No worries, feel free to fix in a follow-up.